src/util/tryCatch.ts

```diff
@@ -11,7 +11,7 @@ function tryCatcher(): any {
   }
 }
 
-export function tryCatch(fn: Function): Function {
+export function tryCatch<T extends Function>(fn: T): T {
   tryCatchTarget = fn;
-  return tryCatcher;
-};
\ No newline at end of file
+  return <any>tryCatcher;
```

Comment:
but the way we're using it, we want `tryCatch` to behave exactly the same as whatever method we give it, return values and everything.  Yes it can return the `errorObject` but there is no way to amend the return type of a method, so it's just a trade off.

It's useful because if for example we decide to add or remove a parameter to something like 

```
takeWhile<T>(predicate: (value: T, index: number) => boolean
```

Then the following method would have compiler errors
```
const result = tryCatch(this.predicate)(value, this.index++);
```